src/Microsoft.AspNetCore.Session/SessionMiddleware.cs

```diff
@@ -35,12 +34,14 @@ public class SessionMiddleware
         /// <param name="dataProtectionProvider">The <see cref="IDataProtectionProvider"/> used to protect and verify the cookie.</param>
         /// <param name="sessionStore">The <see cref="ISessionStore"/> representing the session store.</param>
         /// <param name="options">The session configuration options.</param>
+        /// <param name="sessionKeyGenerator">The <see cref="ISessionKeyGenerator"/> representing the session key generator</param>
         public SessionMiddleware(
             RequestDelegate next,
             ILoggerFactory loggerFactory,
             IDataProtectionProvider dataProtectionProvider,
             ISessionStore sessionStore,
-            IOptions<SessionOptions> options)
+            IOptions<SessionOptions> options,
+            ISessionKeyGenerator sessionKeyGenerator)
```

Comment:
You can't break public API, it doesn't really matter who calls it. We try to not break API in public namespaces unless we're shipping a major release. We even have a tool coming online that will fail the build when you do.